src/ConcreteOperations/intersection.jl

```diff
@@ -89,6 +89,40 @@ function intersection(L1::Line{N}, L2::Line{N}
     end
 end
 
+"""
+    intersection(X::LineSegment{N}, hp::Line{N}
+                ) where {N<:Real}
+Compute the intersection of a line and a line segment.
+### Input
+- `a`  -- LineSegment
+- `b` -- Line
+### Output
+If the sets do not intersect, the result is the empty set.
+Otherwise the result is the singleton or line segment that describes the intersection.
+"""
+
+function intersection(a::LineSegment{N}, b::Line{N}) where {N<:Real}
+  # cast a as line
+  ap = Line(a.p, a.q)
```

Comment:
In this library, `Line` is represented by `ax = b` where `a` is a vector and `b` is a scalar, while `LineSegment` is represented by the two end points (vectors). Hence this constructor does not work. I could not find code to overapproximate a `LineSegment` by a `Line`, so that would need to be added first (can be done in the same PR).

Ideally you would not have to create the `Line` object but would call a helper function that accepts `a` and `b` directly. But this is a micro optimization.

Reply:
Maybe I'm misunderstanding, but there is a line constructor which takes two points through which it passes -- https://github.com/JuliaReach/LazySets.jl/blob/cd6856388521869b89e2700393502cf326527eb3/src/Sets/Line.jl#L50

Reply:
Oh, you are right :+1: I completely forgot about that, sorry!